### PR TITLE
Ensure focus-ring class is applied to contenteditable elements

### PR DESCRIPTION
**Pull request by kloots**
This PR addresses issue #7 and ensures the `focus-ring` class is applied to contenteditable elements. Additionally, it removes the dom-matches polyfill dependency and updates the demo with more use cases.

### CODE REVIEW

File: src/focus-ring.js [hunk 1]
```diff
@@ -36,7 +30,19 @@ document.addEventListener('DOMContentLoaded', function() {
    * @return {boolean}
    */
   function focusTriggersKeyboardModality(el) {
-    return matches(el, keyboardModalityWhitelist) && matches(el, ':not([readonly])');
+    var type = el.type;
+    var tagName = el.tagName.toLowerCase();
```
Comment (jonathantneal) [inline src/focus-ring.js]: You’re only testing `tagName` against `'input'` and `'textarea'`. Therefore, you could safely omit `.toLowerCase()` and test for `'INPUT'` and `'TEXTAREA'` instead.

File: src/focus-ring.js [hunk 2]
```diff
@@ -74,11 +80,13 @@ document.addEventListener('DOMContentLoaded', function() {
    * opening a menu or dialog.
    */
   function onKeyDown() {
-    hadKeyboardEvent = true;
     // `activeElement` defaults to document.body if nothing focused,
     // so check the active element is actually focused.
-    if (matches(document.activeElement, ':focus'))
-      addFocusRingClass(document.activeElement);
+    var activeElement = document.activeElement;
+    if (activeElement.tagName.toLowerCase() == 'body')
```
Comment (jonathantneal) [inline src/focus-ring.js]: Similarly here, there is no browser that will return the tagName of `'BODY'` in lowercase.

Comment (kloots) [inline src/focus-ring.js]: @jonathantneal are you sure? I recall IE returning the declared case if in quirks mode. If you're 100% sure I'm happy to remove the use of `toLowerCase()`

Comment (jonathantneal) [inline src/focus-ring.js]: I am sure.

```html
<body></body><script>document.write(document.body.tagName)</script>
```

Screenshot of Internet Explorer 8 displaying the tagName of body in uppercase

In older IE, there would be an issue with new “unknown” elements (back then, HTML5 elements) returning lowercase. It’s unneeded here, because you’re testing for some of the oldest elements known to web. :)

File: src/focus-ring.js [hunk 3]
```diff
@@ -31,7 +31,7 @@ document.addEventListener('DOMContentLoaded', function() {
    */
   function focusTriggersKeyboardModality(el) {
     var type = el.type;
-    var tagName = el.tagName.toLowerCase();
+    var tagName = el.tagName;
 
     if (tagName == 'input' && inputTypesWhitelist[type] && !el.readonly)
```
Comment (jonathantneal) [inline src/focus-ring.js]: Be sure to test these changes. I’m not sure `'input'` will work because it should now be uppercase?

File: dist/focus-ring.js [hunk 4]
```diff
@@ -253,7 +200,19 @@ document.addEventListener('DOMContentLoaded', function() {
...
+    var type = el.type;
+    var tagName = el.tagName.toLowerCase();
+
+    if (tagName == 'input' && inputTypesWhitelist[type] && !el.readonly)
+      return true;
+
+    if (tagName == 'textarea' && !el.readonly)
+      return true;
+
+    if (el.contentEditable)
```
Comment (jonathantneal) [inline dist/focus-ring.js]: The possible values of `contentEditable` are `'inherit'`, `'true'`, or `'false'`.

Comment (kloots) [inline dist/focus-ring.js]: Fixed.

File: src/focus-ring.js (hunk 3 above)
Comment (kloots) [inline src/focus-ring.js]: Fixex

**Comment by kloots** [comment]
@alice, any thoughts on this PR?

File: src/focus-ring.js [hunk 5]
```diff
@@ -74,11 +80,13 @@ document.addEventListener('DOMContentLoaded', function() {
...
-    hadKeyboardEvent = true;
     // `activeElement` defaults to document.body if nothing focused,
     // so check the active element is actually focused.
-    if (matches(document.activeElement, ':focus'))
-      addFocusRingClass(document.activeElement);
+    var activeElement = document.activeElement;
+    if (activeElement.tagName == 'BODY')
+      return;
+
+    hadKeyboardEvent = true;
```
Comment (alice) [inline src/focus-ring.js]: Why was this moved below the check?

Comment (kloots) [inline src/focus-ring.js]: Because the more I thought about it, we don't want to set that to true if focus hasn't moved to a focusable control. This way we can use `hadKeyboardEvent` to determine if a focus moved to a focusable element as a result of keyboard navigation.

Comment (alice) [inline src/focus-ring.js]: `hasKeyboardEvent` is not intended for that use, though. I think this could cause a bug if a keyboard event triggered a focus move (for example, activating a button which caused a piece of modal UI to appear). Please revert this change.

Comment (kloots) [inline src/focus-ring.js]: Sorry, I'm not 100% following you regarding the bug that could result from this change. Mind explaining in great detail?

Comment (alice) [inline src/focus-ring.js]: Ah - I had it backwards. However, since the keyboard event precedes the focus event, this is checking where focus is coming _from_, rather than going _to_, isn't it?

Comment (kloots) [inline src/focus-ring.js]: No worries at all. To answer your question: it's the other way around. The fact the keyboard event precedes the focus event actually works to our advantage here because we know the event sequence is: keydown then focus, and therefore the focus handler can key off of `hadKeyboardEvent` to know focus happened as a result of keyboard, and then add the focus ring class.

Comment (alice) [inline src/focus-ring.js]: Right: so at the point where this fires, `activeElement` is the element which is about to _lose_ focus, isn't it? So why _not_ set `hadKeyboardEvent` based on that?

Comment (kloots) [inline src/focus-ring.js]: Ah, yes! Now I see what you're saying. Will change.

**Comment by alice** [comment]
👍
